### PR TITLE
Introduce "chewing" to reduce the volume of low density foods

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary
SUMMARY: Balance "Adjusts hunger and stomach contents balance in a few small ways"

#### Purpose of change
As always we run into a few problems with certain foods and their impact on satiety. Popcorn, for example, has extraordinarily high satiety due to its large volume.

#### Describe the solution
When you `E`at food you now get a minigame where you see an ASCII art presentation of a mouth and morsel of food, and must repeatedly press E followed by S to chew and swallow the food. Too fast and you could bite your tongue, for a morale penalty. Not enough and you could choke! Mutations and attributes affect chewing speed and efficiency. There is also a new "speed eater" proficiency for chewing quickly that reduces your tongue bite chance.

##### Mastication
Just kidding.

IIn short, this adds a masticate function to condense bulky, low-density foods down to the density of water before adding them to the stomach. The process is:

1. Calculate dry weight and dry volume of the food
2. If the density of the dried food is <1, set the dry volume to be equal to the dry weight so that the density = 1.
3. Use this new masticated_volume to be the food volume anywhere it counts. Add satiety calculations as usual, so if the food is calorie dense by the gram we apply that density to the crunched volume, not the fluffy one.

This has fixed the popcorn satiety problem. Popcorn now adds about 25mL to your stomach instead of about 800 mL. It should work similarly well for potato chips and a lot of other dry snacks. It also solves berries and a lot of the less filling vegetables like lettuce. Jerky too.

#### Describe alternatives you've considered
Some version of this is needed. The drop to 1g/mL is pretty arbitrary but I think it works. For some of our satiety changes, working in macronutrients would be a much more complex way of achieving what we want, but I think actually we're going to get something similar by the end of this without all the JSON work.

I am not sure doing the volume of the dry food is the best way to do this, but at some point we need to remove the water from the food, and doing it after we apply all these maths to the food has potential for some unexpected side effects I think. There are other, mathier ways to do it and if we get some weird outliers during testing I might have to try those.

#### Testing
Popcorn stomach volume drops to 25mL from over 800.
Lettuce adds 38mL to the stomach, compared to 250mL before.
Blueberries add about 150mL to the stomach, compared to 445mL before.
Bratwurst, spaghetti, and chicken soup stomach volumes remain pretty consistent with what would be expected before the change.

Bratwurst before:
image

Bratwurst after:
image

After 1 jerky before:
image

After 1 jerky now:
image

Boring sandwich, a previous problem food, before:
image

Boring sandwich now:
image

Deluxe sandwich before, strangely lower than boring sadnwich:
image

Deluxe sandwich now, I need to figure out why this has a higher volume though:
image

Pemmican before:
image

Pemmican after:
image

I still need to review the code a bit further. I am not sure why deluxe sandwich volume increased. For the rest, the foods that were previously high satiety remain high but maybe a little toned down.

#### Additional context
I was originally going to adjust stomach emptying too, but I think that better go in a separate PR as it has more potential for issues.

I thought I had somehow broken weary tests here but that happened earlier. Nonetheless, I fixed them here.

### CODE REVIEW

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/height-should-affect-stomach-content-and-calorie-storage/25078/3

File: src/consumption.cpp [hunk 1]
```diff
@@ -1245,10 +1264,7 @@ int Character::compute_calories_per_effective_volume( const item &food,
     } else {
         kcalories = compute_effective_nutrients( food ).kcal;
     }
-    units::volume water_vol = ( food.get_comestible()->quench > 0 ) ? food.get_comestible()->quench *
-                              5_ml : 0_ml;
-    // Water volume is ignored.
-    units::volume food_vol = food.volume() - water_vol * food.count();
+    units::volume food_vol = masticated_volume( food );
```
Comment (I-am-Erk) [inline src/consumption.cpp:1267]:
```suggestion
    units::volume food_vol = masticated_volume( food ) * food.count();
```

**Comment by maibock-ell** [comment]
For the most part, I am fine with these changes, but the minigame seems a little out of place. It would be interesting at first, but would quickly become a chore for players.

**Comment by I-am-Erk** [comment]
It's good to know I got at least one person.

image

**Comment by maibock-ell** [comment]
Oh my god! You got me there, hahah. You know, with all these 'realism' PRs, I would not be surprised.

File: tests/comestible_test.cpp [hunk 2]
```diff
@@ -231,8 +231,8 @@ TEST_CASE( "effective food volume and satiety", "[character][food][satiety]" )
     // If kcal per gram > 3.0, return sqrt( 3 * kcal / gram )
     expect_ratio = std::sqrt( 3.0f * 202 / 30 );
     CHECK( u.compute_effective_food_volume_ratio( nuts ) == Approx( expect_ratio ).margin( 0.01f ) );
-    CHECK( u.compute_calories_per_effective_volume( nuts ) == 642 );
-    CHECK( satiety_bar( 642 ) == "<color_c_light_green>|||</color>.." );
+    CHECK( u.compute_calories_per_effective_volume( nuts ) == 1498 );
+    CHECK( satiety_bar( 1498 ) == "<color_c_green>||||\\</color>.." );
```
Comment (kevingranade) [inline tests/comestible_test.cpp:235]:
```suggestion
    CHECK( satiety_bar( 1498 ) == "<color_c_green>||||\\</color>" );
```

**Comment by I-am-Erk** [comment]
Sweet

**Pull request merged**